doc/sample/getiteminfo.txt

```diff
@@ -13,28 +16,170 @@ prontera,156,179,6	script	test_getiteminfo	117,{
 	case 1:
 		mes "Please enter an item ID.";
 		input .@value;
-
+		if(getiteminfo(.@value,ITEMINFO_ID) == -1) { // Item doesn't exist, so end the script
+			mes "Unknown Item, please type the ID of an existing item.";
+			end;
+		}
 		// This line uses an INTERNAL function of your client to show item name by its ID!
 		// ^nItemID^XXXX -> Item Name
 		mes "Item ID: " + .@value + " ^nItemID^" + .@value;
-
-		mes "Current item info:";
-		for(.@id = 0; .@id <= 18; .@id++)
-			mes " getiteminfo(" + .@value + "," + .@id + ") = " + getiteminfo(.@value,.@id);
-		close;
+		break;
 	case 2:
 		mes "Please enter an english item name.";
 		input .@value$;
+		if(getiteminfo(.@value$,ITEMINFO_ID) == -1) { // Item doesn't exist, so end the script
+			mes "Unknown Item, please type the english name of an existing item.";
+			end;
+		}
 		mes "english item name: " + .@value$;
 		break;
 	case 3:
 		mes "Please enter an aegis item name.";
 		input .@value$;
+		if(getiteminfo(.@value$,ITEMINFO_ID) == -1) { // Item doesn't exist, so end the script
+			mes "Unknown Item, please type the aegis name of an existing item.";
+			end;
+		}
 		mes "aegis item name: " + .@value$;
 		break;
 	}
+	if(!.@value && .@value$ != "") // To prevent to duplicate the loop
+		.@value = getiteminfo(.@value$,ITEMINFO_ID);
```

Comment:
You should do that in the respective case above, since you already look up the Item ID there, to check if the item exists anyway.

Reply:
So writing it twice instead of once? Sounds like an unoptimized way of doing it tbh.
Also I'm only, like mentioned, using .@value as main variable to not duplicate the for loop as it was previously.